app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
+    for (var i = 0; i < $scope.hits.length - 1; i++) {
+      $scope.hits[i] = calcDamage($scope.hits[i].hit);
+    }
+    if (!skipSerialization) {
+      serializeData();
+    }
+  };
+
+  $scope.updateHitTable = function (index) {
```

Comment:
I think updateHitRow is more accurate. If the row is the current hanging row it makes a new one to replace itself, but that's not an 'update' its an 'insert' and is part of the row update. I mean, I guess this is an indication that 'addNewRowIfNeeded()' should be its own method or something, but I think its fine as is.